app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,16 @@
+class SessionsController < ApplicationController
+  def create
+    if user = User.find_by(email: params[:email])
+      if user.authenticate(params[:password])
+        session[:user_id] = user.id
+        redirect_to '/user/dashboard'
+      else
+        flash[:error] = "Credentials are incorrect"
+        redirect_to '/'
+      end
+    else
+      flash[:error] = "Credentials are incorrect"
+      redirect_to '/'
+    end
+  end
```

Comment:
A little lenghty. Forgot to run RuboCop to refactor into smaller method.

Reply:
We can refactor that on next branch